### ISSUE TITLE
Platform independent PowerShell: unable to resolve the correct dataLocalDir

### ISSUE THREAD

**Issue by WordlessEcho** [bluefireoly/pacmc#8]
```
$ pacmc init
Exception in thread "main" java.lang.ExceptionInInitializerError
        at net.axay.pacmc.commands.Init$run$1.invokeSuspend(Init.kt:25)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: org.kodein.db.leveldb.LevelDBException: NotFound: D:\files\app\games\Minecraft\.minecraft\null\axay\pacmc\data\db1/LOCK: ?????????
        at org.kodein.db.leveldb.jni.Native.dbOpen(Native Method)
        at org.kodein.db.leveldb.jni.LevelDBJNI$Factory.open(LevelDBJNI.kt:29)
        at org.kodein.db.leveldb.jvm.LevelDBJvm.open(LevelDBJvm.kt)
        at org.kodein.db.impl.kv.AbstractKeyValueDBFactory.open(AbstractKeyValueDBFactory.kt:20)
        at org.kodein.db.impl.kv.AbstractKeyValueDBFactory.open(AbstractKeyValueDBFactory.kt:12)
        at org.kodein.db.impl.data.AbstractDataDBFactory.open(AbstractDataDBFactory.kt:20)
        at org.kodein.db.impl.data.AbstractDataDBFactory.open(AbstractDataDBFactory.kt:12)
        at org.kodein.db.impl.model.AbstractModelDBFactory.open(AbstractModelDBFactory.kt:29)
        at org.kodein.db.impl.model.AbstractModelDBFactory.open(AbstractModelDBFactory.kt:9)
        at org.kodein.db.impl.AbstractDBFactory.open(AbstractDBFactory.kt:25)
        at org.kodein.db.impl.AbstractDBFactory.open(AbstractDBFactory.kt:10)
        at org.kodein.db.impl.DefaultKt.open(default.kt:7)
        at net.axay.pacmc.storage.DatabaseKt.<clinit>(Database.kt:14)
        ... 7 more
```

Pacmc version:
```powershell
$ scoop update pacmc
pacmc: 0.2.1 (latest version)
```

JDK version:
```
$ java --version
openjdk 16.0.1 2021-04-20
OpenJDK Runtime Environment (build 16.0.1+9-24)
OpenJDK 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
```

**Comment by WordlessEcho** [comment]
Here is `Values.projectDirectories` and `Values.projectDirectories.dataLocalDir` I got:
```
ProjectDirectories (Windows 10):
  projectPath   = 'axay\pacmc'
  cacheDir      = 'null\axay\pacmc\cache'
  configDir     = 'PowerShell 7.1.3\axay\pacmc\config'
  dataDir       = 'PowerShell 7.1.3\axay\pacmc\data'
  dataLocalDir  = 'null\axay\pacmc\data'
  preferenceDir = 'PowerShell 7.1.3\axay\pacmc\config'
  runtimeDir    = 'null'
```
```
null\axay\pacmc\data
```

**Comment by WordlessEcho** [comment]
I got same exception with `cmd.exe` but `Values.projectDirectories` and `Values.projectDirectories.dataLocalDir` is fine. Seems like that this exception not about the path.
```
ProjectDirectories (Windows 10):
  projectPath   = 'axay\pacmc'
  cacheDir      = 'C:\Users\echo\AppData\Local\axay\pacmc\cache'
  configDir     = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
  dataDir       = 'C:\Users\echo\AppData\Roaming\axay\pacmc\data'
  dataLocalDir  = 'C:\Users\echo\AppData\Local\axay\pacmc\data'
  preferenceDir = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
  runtimeDir    = 'null'
```
```
C:\Users\echo\AppData\Local\axay\pacmc\data
```

**Comment by WordlessEcho** [comment]
Workaround: create floaders `.\axay\pacmc\data` on `C:\Users\echo\AppData\Local`.

**Comment by bluefireoly** [comment]
Thank you for your report, I will boot into Windows and test it by myself - if that does not give me the same result as you got, I will have a look into the code which resolves the `dataLocalDir`.

**Comment by bluefireoly** [comment]
Did you test it on both Windows 11 and Windows 10?

**Comment by bluefireoly** [comment]
I have a hard time seeing where you executed what.

Where exactly did you get these results?
> ```
> ProjectDirectories (Windows 10):
>   projectPath   = 'axay\pacmc'
>   cacheDir      = 'null\axay\pacmc\cache'
>   configDir     = 'PowerShell 7.1.3\axay\pacmc\config'
>   dataDir       = 'PowerShell 7.1.3\axay\pacmc\data'
>   dataLocalDir  = 'null\axay\pacmc\data'
>   preferenceDir = 'PowerShell 7.1.3\axay\pacmc\config'
>   runtimeDir    = 'null'
> ```

And in which case did you get these correct results?

> ```
> ProjectDirectories (Windows 10):
>   projectPath   = 'axay\pacmc'
>   cacheDir      = 'C:\Users\echo\AppData\Local\axay\pacmc\cache'
>   configDir     = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
>   dataDir       = 'C:\Users\echo\AppData\Roaming\axay\pacmc\data'
>   dataLocalDir  = 'C:\Users\echo\AppData\Local\axay\pacmc\data'
>   preferenceDir = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
>   runtimeDir    = 'null'
> ```

**Comment by WordlessEcho** [comment]
> Did you test it on both Windows 11 and Windows 10?

Yes.

> I have a hard time seeing where you executed the two examples you gave.
> 
> Where exactly did you get these results?
> 
> > ```
> > ProjectDirectories (Windows 10):
> >   projectPath   = 'axay\pacmc'
> >   cacheDir      = 'null\axay\pacmc\cache'
> >   configDir     = 'PowerShell 7.1.3\axay\pacmc\config'
> >   dataDir       = 'PowerShell 7.1.3\axay\pacmc\data'
> >   dataLocalDir  = 'null\axay\pacmc\data'
> >   preferenceDir = 'PowerShell 7.1.3\axay\pacmc\config'
> >   runtimeDir    = 'null'
> > ```
> 
> And in which case did you get these correct results?
> 
> > ```
> > ProjectDirectories (Windows 10):
> >   projectPath   = 'axay\pacmc'
> >   cacheDir      = 'C:\Users\echo\AppData\Local\axay\pacmc\cache'
> >   configDir     = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
> >   dataDir       = 'C:\Users\echo\AppData\Roaming\axay\pacmc\data'
> >   dataLocalDir  = 'C:\Users\echo\AppData\Local\axay\pacmc\data'
> >   preferenceDir = 'C:\Users\echo\AppData\Roaming\axay\pacmc\config'
> >   runtimeDir    = 'null'
> > ```

First one I tested on **[PowerShell 7.1.3](https://github.com/PowerShell/PowerShell/releases/tag/v7.1.3)**. Second one I tested on **system integrated PowerShell**.

**Comment by bluefireoly** [comment]
Ah I see, as I am not a Windows user, can you give me a hint how I can test it on the not system integrated PowerShell? I was only testing it in the PowerShell which is a default profile in Windows Terminal.

**Comment by WordlessEcho** [comment]
1. Install [PowerShell-7.1.3-win-x64.msi](https://github.com/PowerShell/PowerShell/releases/download/v7.1.3/PowerShell-7.1.3-win-x64.msi) on your Windows (or [x86](https://github.com/PowerShell/PowerShell/releases/download/v7.1.3/PowerShell-7.1.3-win-x86.msi) if you are using 32 bit).
2. After installation finished, reopen Windows Terminal and click down arrow in tab bar. **Windows PowerShell** is your system integrated PowerShell. PowerShell is the PS 7.1.3 you just installed.
![image](https://user-images.githubusercontent.com/12007025/125616382-0268b75e-5c70-4b67-ad0a-b9855f4e9b63.png)

**Comment by bluefireoly** [comment]
Thank you, I was able to reproduce the problem, will try to fix this now.

**Comment by bluefireoly** [comment]
Fixed with 1f3d92b and 309028c

Also added a new command called `pacmc info` (967a06e), you can check if the paths are detected correctly using this command.

**Comment by bluefireoly** [comment]
The fix is available in version `0.2.2`, you have to wait some hours until scoop picks up that update.

**Comment by WordlessEcho** [comment]
I have deleted the `C:\Users\<USERNAME>\AppData\Local\axay` and test again. Works great for me, thx!

```
$ .\pacmc.bat info
pacmc version 0.2.2

dataLocalDir: C:\Users\echo\AppData\Local\axay\pacmc\data
databaseDir: C:\Users\echo\AppData\Local\axay\pacmc\data\db1
databaseDir (canonical): C:\Users\echo\AppData\Local\axay\pacmc\data\db1

Your OS: WINDOWS
Java version: 16.0.1+9-24
```
